app/controllers/patients_controller.rb

```diff
@@ -249,53 +249,170 @@ def update
-    # id exists. Patient.exists?(nil) => false
-    redirect_to(root_url) && return unless Patient.where(purged: false).exists?(new_hoh_id.to_i)
+    # ----- Error Checking -----
 
-    patients_to_update = household_ids + [current_patient_id]
+    # Check to make sure user has access to update this record.
+    unless current_user_patients.exists?(current_patient_id)
+      error_message = 'Remove from household action failed: User does not have permissions to update current monitoree.'
+      render(json: { error: error_message }, status: :forbidden) && return
+    end
 
-    # Make sure all household ids are within jurisdiction
-    redirect_to(root_url) && return if patients_to_update.any? do |patient_id|
-      !current_user_patients.exists?(patient_id)
+    # If the current patients is a HoH, they can't be removed.
+    if current_patient.head_of_household
+      error_message = 'Remove from household  action failed: Monitoree is a head of household. Please refresh.'
+      render(json: { error: error_message }, status: :bad_request) && return
     end
 
-    # Change all of the patients in the household, including the current patient to have new_hoh_id as the responder
-    current_user_patients.where(id: patients_to_update).each do |patient|
-      patient.update!(responder_id: new_hoh_id)
+    # ----- Record Updates -----
+    old_hoh = current_user.get_patient(current_patient.responder_id)
+
+    # Update the record
+    updated = current_patient.update(responder_id: current_patient.id)
+
+    if !updated
+      error_message = 'Remove from household action failed: Monitoree was unable to be be updated.'
+      render(json: { error: error_message, status: :forbidden }) && return
+    else
+      # Create history item for old HoH
+      comment = "User removed dependent monitoree with ID #{current_patient.id} from the household. #{old_hoh.first_name} #{old_hoh.last_name}"\
+                ' will no longer be responsible for handling their reporting.'
+      History.monitoring_change(patient: old_hoh, created_by: current_user.email, comment: comment)
+
+      # Create history item on current patient
+      comment = "User removed monitoree from a household. Monitoree with ID #{old_hoh.id} will"\
+                ' no longer be responsible for handling their reporting.'
+      History.monitoring_change(patient: current_patient, created_by: current_user.email, comment: comment)
+    end
+  end
+
+  # Changes the HoH of a household
+  def update_hoh
+    new_hoh_id = params.permit(:new_hoh_id)[:new_hoh_id]&.to_i
+    current_patient_id = params.permit(:id)[:id]
+    household_ids = params[:household_ids] || []
+
+    current_patient = current_user.get_patient(current_patient_id)
+    new_hoh = current_user.get_patient(new_hoh_id)
+    current_user_patients = current_user.patients
+
+    # NOTE: This includes the new HoH
+    patient_ids_to_update = household_ids + [current_patient_id]
+
+    # ----- Error Checking -----
```

Comment:
I think another edge case to watch out for is if another user happens to add a new household member between the time this user (the one submitting this request) started viewing the page and the time this request is submitted and I'm not sure what the expected behavior for that case would be? Here are some options I can think of:
- return an error explaining the situation and don't update anything
- continue to update HoH for all members of the household including the new household member(s) that this user didn't see in the UI before, but was added by another user (I'm guessing that's what users might actually want but could be potentially confusing?)

I think we just have to avoid the situation where we only update HoH for the household members from `household_ids` when `household_ids` doesn't actually include all household members because I think that could potentially lead to the original HoH becoming both an HoH and dependent? This kind of makes me think we should just always use/fetch the list of household members from the back end instead of the client to avoid this kind of situation. Thoughts?

Reply:
That's exactly what I'm thinking to resolve Mike's issue above. Will work on this solution.

Reply:
Fixed here: d18e3ff